base/libgit2/libgit2.jl

```diff
@@ -644,6 +686,13 @@ function snapshot(repo::GitRepo)
     State(head, index, work)
 end
 
+"""
+    restore(s::State, repo::GitRepo)
+
+Return a repository `repo` to a previous `State` `s`, for
+example the HEAD of a branch before a merge attempt. `s`
+can be generated using the [`snapshot`](@ref) function.
+"""
```

Comment:
Probably not necessary for this PR but I think `restore` would be a good candidate for a signature change to `restore!(repo, state)`, since `repo` is modified in the process.